File: backend/src/main/java/com/project2/global/security/CustomAuthenticationSuccessHandler.java
package com.project2.global.security;

import java.io.IOException;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
import org.springframework.stereotype.Component;

import com.project2.domain.member.entity.Member;
import com.project2.domain.member.service.AuthService;

import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.servlet.http.HttpSession;
import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class CustomAuthenticationSuccessHandler implements AuthenticationSuccessHandler {

	private final Rq rq;
	private final AuthService authService;

	@Value("${custom.url.front-url}")
	private String siteFrontUrl;

	@Override
	public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
		Authentication authentication) throws IOException, ServletException {
		HttpSession session = request.getSession();

		String redirectUrl = (String)session.getAttribute("redirectUrl");
		if (redirectUrl == null) {
			redirectUrl = siteFrontUrl;
		}
		session.removeAttribute("redirectUrl");

		Member member = rq.getActor();
		String accessToken = authService.genAccessToken(member);
		String refreshToken = authService.genRefreshToken(member);

		rq.addCookie("accessToken", accessToken, false);
		rq.addCookie("refreshToken", refreshToken, true);

		response.sendRedirect(redirectUrl);
	}
}

File: backend/src/main/kotlin/com/project3/global/security/CustomAuthenticationSuccessHandler.kt
package com.project3.global.security

import com.project3.domain.member.entity.Member
import com.project3.domain.member.service.AuthService
import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import jakarta.servlet.http.HttpSession
import org.springframework.beans.factory.annotation.Value
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AuthenticationSuccessHandler
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class CustomAuthenticationSuccessHandler(
        private val rq: Rq,
        private val authService: AuthService
) : AuthenticationSuccessHandler {

    @Value("\${custom.url.front-url}")
    private lateinit var siteFrontUrl: String

    @Throws(IOException::class, ServletException::class)
    override fun onAuthenticationSuccess(
            request: HttpServletRequest,
            response: HttpServletResponse,
            authentication: Authentication
    ) {
        val session: HttpSession = request.session
        var redirectUrl = session.getAttribute("redirectUrl") as String?
        if (redirectUrl == null) {
            redirectUrl = siteFrontUrl
        }
        session.removeAttribute("redirectUrl")

        val member: Member = rq.getActor()
        val accessToken = authService.genAccessToken(member)
        val refreshToken = authService.genRefreshToken(member)

        rq.addCookie("accessToken", accessToken, false)
        rq.addCookie("refreshToken", refreshToken, true)

        response.sendRedirect(redirectUrl)
    }
}
